sdks/python/apache_beam/coders/coder_impl.py

```diff
@@ -253,8 +258,8 @@ def decode(self, encoded):
 NONE_TYPE = 0
 INT_TYPE = 1
 FLOAT_TYPE = 2
-STR_TYPE = 3
-UNICODE_TYPE = 4
+BYTES_TYPE = 3
+TEXT_TYPE = 4
```

Comment:
I would leave this as UNICODE_TYPE. TEXT_TYPE is ambiguous wrt. Python 2 vs. 3.

Reply:
Also, any renaming here should be reflected in the .pxd file. That may account for some of the perf issues. 

Running "python -m cython -a apache_beam/coders/coder_impl.py" could be useful for debugging this too.